importers/src/main/groovy/whelk/importer/WhelkCopier.groovy

```diff
@@ -24,20 +24,42 @@ class WhelkCopier {
     }
 
     void run() {
+        TreeSet<String> alreadyImportedIDs = new TreeSet<>()
+
         for (id in recordIds) {
-            def doc = source.getDocument(id)
+            def doc
+            if (id.contains("/")) {
+                doc = source.storage.getDocumentByIri(id)
+            }
+            else
+                doc = source.getDocument(id)
+            if (doc == null) {
+                System.err.println("Could not load document with ID: $id");
+                continue
+            }
+            doc.baseUri = source.baseUri
+
             // this links to:
             for (relDoc in selectBySqlWhere("""id in (select dependsonid from lddb__dependencies where id = '${id}')""")) {
```

Comment:
Should we get more than one level of dependencies? (PostgreSQLComponent.followDependencies)

Reply:
Perhaps, but that would then be a separate issue to this :)